system/ee/ExpressionEngine/Addons/member/mod.member.php

```diff
@@ -1601,6 +1601,26 @@ public function login_form()
         return $res;
     }
 
+    /**
+     * Allows for validation of username and password using Ajax
+     * @return void
+     */
+    public function login_validate()
+    {
+        ee()->load->library('Auth');
+        $verify = ee()->auth->verify();
+        $reasons = [];
+        if(!$verify) {
+            $reasons = ee()->auth->errors;
+        }
+
+        echo json_encode([
+            'result' => ($verify ? 'success' : 'fail'),
+            'reason' => $reasons,
+            'username' => ee()->input->get('username'),
+        ]);
```

Comment:
It is better to use `ee()->output->send_ajax_response` as it sends the correct header and assures there's no profiler present in AJAX response, if it's enabled

Also not real issue, but `result => fail` is not being used in EE, more common are things `'messageType' => 'error',` or `'status' => 'error',`. As well as `reason` - could it be `error`? Again, this will not create any issues, just trying to keep it more consistent with what we have in other places - let me know if you have specific reason to keep them this way.

I will also make the PR draft until the action record part and the docs are ready

Reply:
Nope, no reason whatsoever for it; was just a carry over from previous implementations. What you think of something like this:

```php
        ee()->output->send_ajax_response([
            'messageType' => ($verify ? 'success' : 'error'),
            'errors' => $reasons,
            'username' => ee()->input->get('username'),
        ]);
```

Reply:
... maybe `messages` in lieu of `errors`...